components/raftstore-v2/src/batch/store.rs

```diff
@@ -524,12 +527,20 @@ impl<EK: KvEngine, ER: RaftEngine> StoreSystem<EK, ER> {
         read_runner.set_snap_mgr(snap_mgr.clone());
         let read_scheduler = workers.async_read.start("async-read-worker", read_runner);
 
+        let stats_monitor = PdStatsMonitor::new(
```

Comment:
The whole thing is a hot mess right now, everybody can talk to anybody. Moving it out is a first step to make the boundaries between actors clearer. We can actually see a case in v1, the pd scheduler is sent into pd runner (to create the stats monitor), and as a result someone misuses the pd scheduler when handling normal pd task.

Reply:
> everybody can talk to anybody

I don't see what is wrong. Actors of cause can talk to any actors. And I don't see how it's related to `PdStatsMonitor`.

> someone misuses the pd scheduler when handling normal pd task.

I don't get it. What kind of misuse?

The point of keeping it inside pd worker is to avoid leaking abstraction. Nobody except pd runner cares about PdStatsMonitor, so it should build the monitor directly.

Reply:
Let's have it your way, I don't care about it that much.